manifests/agent/service.pp

```diff
@@ -1,17 +1,49 @@
-class check_mk::agent::service {
-  if ! defined(Service['xinetd']) {
-    if $::operatingsystem == 'Debian' and
-    versioncmp($::operatingsystemmajrelease, '7') == 0 {
+# Class: check_mk::agent::service
+#
+# @summary Manages the xinetd service
+#
+# @api private
+#
+class check_mk::agent::service (
+  Boolean $use_xinetd = $check_mk::agent::use_xinetd,
+)
+{
+  if $use_xinetd {
+    ensure_packages(['xinetd'])
+    Package['xinetd'] ~> Service['xinetd']
+
+    service { 'check_mk.socket':
```

Comment:
did you consider not deploying the systemd config if xinted is used? I don't have an opinion here, just curious.

Reply:
The package installs this systemd file (even on systems without systemd).  This module manages the drop-in files.